source/2_platform/rex_directx/src/directx_renderer.cpp

```diff
@@ -0,0 +1,122 @@
+    bool is_depth_0_to_1()
+    {
+      return true;
+    }
+
+    namespace backend
+    {
+      struct DirectXContext
+      {
+        wrl::com_ptr<ID3D12Device> device = nullptr;
+      };
+
+      DirectXContext g_ctx; // NOLINT(fuchsia-statically-constructed-objects, cppcoreguidelines-avoid-non-const-global-variables)
+
+      //-------------------------------------------------------------------------
+      bool initialize()
+      {
+        rsl::unique_ptr<dxgi::Factory> factory = dxgi::Factory::create();
+
+        dxgi::AdapterManager adapter_manager(factory.get(), HighestVramGpuScorer());
+        const dxgi::Adapter* adapter = static_cast<const dxgi::Adapter*>(adapter_manager.selected()); // NNOLINT(cppcoreguidelines-pro-type-static-cast-downcast)
+
+        const FeatureLevelInfo feature_level_info = query_feature_level(adapter->c_ptr());
+
+        if(FAILED(D3D12CreateDevice(adapter->c_ptr(), static_cast<D3D_FEATURE_LEVEL>(feature_level_info.level), IID_PPV_ARGS(&g_ctx.device))))
+        {
+          if(adapter_manager.load_software_adapter(factory.get()))
+          {
+            REX_ERROR(LogDirectX, "Could not create software adapter");
+            REX_ERROR(LogDirectX, "Failed to create DX12 Device");
+            return false;
+          }
+          else
+          {
+            const dxgi::Adapter* software_adapter = static_cast<const dxgi::Adapter*>(adapter_manager.software()); // NNOLINT(cppcoreguidelines-pro-type-static-cast-downcast)
+            if(FAILED(D3D12CreateDevice(software_adapter->c_ptr(), static_cast<D3D_FEATURE_LEVEL>(feature_level_info.level), IID_PPV_ARGS(&g_ctx.device))))
+            {
+              REX_ERROR(LogDirectX, "Failed to create DX12 Device");
+              return false;
+            }
+          }
+        }
+        else
+        {
+          REX_LOG(LogDirectX, "D3D12 Device Created!");
+        }
+
+        const ShaderModelInfo shader_model_info = query_shader_model_version(g_ctx.device.Get());
+
+        directx::g_renderer_info.shader_version = shader_model_info.name;
+        directx::g_renderer_info.api_version    = feature_level_info.name;
+        directx::g_renderer_info.adaptor        = adapter->description().name;
+        directx::g_renderer_info.vendor         = adapter->description().vendor_name;
+
+        return true;
+      }
+
+      //-------------------------------------------------------------------------
+      void shutdown()
+      {
+        // Nothing to implement
+      }
+
+      //-------------------------------------------------------------------------
+      void new_frame()
+      {
+        // Nothing to implement
+      }
+
+      //-------------------------------------------------------------------------
+      void end_frame()
+      {
+        // Nothing to implement
+      }
+
+      //-------------------------------------------------------------------------
+      void clear() {}
```

Comment:
nothing to implement statement? 
I know it'll be implement in the future, but just to be consistent with the rest